vhdl_lang/src/config.rs

```diff
@@ -278,6 +283,34 @@ impl Config {
     }
 }
 
+fn substitute_environment_variables(s: &str, replace: &HashMap<String, String>) -> String {
+    let mut result = String::new();
+
+    let mut left = s;
+    while let Some(start) = left.find('$') {
+        // keep non-env var piece as-is
+        result.push_str(&left[..start]);
+        left = &left[start + 1..];
+
+        // replace env var
+        let env_name_len = left
+            .find(|c: char| !(c == '_' || c.is_ascii_alphanumeric()))
+            .unwrap_or(left.len());
+        let env_name = &left[..env_name_len];
+
+        let replacement = replace.get(env_name).map(String::as_ref).unwrap_or("");
```

Comment:
Same as above, I would prefer if this is communicated to the user.

Reply:
I agree, I'll add a warning.

Reply:
There's a mix of two error handling mechanisms used in this file:
* `Config::from_str` can report a single error through the `Result` it returns, which also marks a total failure to load the file.
* Most other functions report errors or warnings through `MessageHandler` and keep going with a best effort.
Which option should be used for these env var errors? I think the latter fits better but that requires adding a `messages` parameter to `Config::from_str`, is that okay?

Reply:
Given that all other issues with the `vhdl_ls.toml` file also cause an error I would prefer the first option you proposed.